hack/generated/pkg/armclient/template_client.go

```diff
@@ -311,14 +312,28 @@ func (atc *AzureTemplateClient) HeadResource(ctx context.Context, id string, api
 	}
 }
 
-func MakeArmResourceId(subscriptionId string, segments ...string) (string, error) {
-	// There should be an even number of segments
-	if len(segments)%2 != 0 {
-		return "", errors.Errorf("expected even number of ARM resource ID segments, got: %d", len(segments))
+func MakeSubscriptionScopeARMId(subscription string, provider string, params ...string) string {
+	if len(params) == 0 {
+		panic("At least 2 params must be specified")
+	}
+	if len(params)%2 != 0 {
+		panic("ARM Id params must come in resourceKind/name pairs")
+	}
+
+	suffix := strings.Join(params, "/")
+
+	return fmt.Sprintf("/subscriptions/%s/providers/%s/%s", subscription, provider, suffix)
+}
+
+func MakeResourceGroupScopeARMId(subscription string, resourceGroup string, provider string, params ...string) string {
+	if len(params) == 0 {
+		panic("At least 2 params must be specified")
+	}
+	if len(params)%2 != 0 {
+		panic("ARM Id params must come in resourceKind/name pairs")
 	}
 
-	start := "/subscriptions/" + subscriptionId
-	remaining := strings.Join(segments, "/")
+	suffix := strings.Join(params, "/")
 
-	return start + "/" + remaining, nil
+	return fmt.Sprintf("/subscriptions/%s/resourceGroups/%s/providers/%s/%s", subscription, resourceGroup, provider, suffix)
```

Comment:
Can/should `MakeResourceGroupScopeARMId()` delegate to `MakeSubscriptionScopeARMId()`?

Reply:
It can't without changing the way they're both writter, as `MakeSubscriptionScopeARMId` explicitly forbids tacking things after `/subscriptions/<>/` but before the `providers` section. If I changed `MakeSubscriptionScopeARMId` to allow that it wouldn't be as safe because it'd allow other (incorrect) formats too presumably.

We could probably refactor an internal method that both delegated to, but doing so would probably end up basically just being a slightly less generic `strings.Join`, so it doesn't seem worth it?

Reply:
Fair enough.